File: LoginDagger/app/src/main/java/com/charlye934/loginddagger/root/ApplicationComponent.java
package com.charlye934.loginddagger.root;

import com.charlye934.loginddagger.http.TwitchModule;
import com.charlye934.loginddagger.http.twitch.Twitch;
import com.charlye934.loginddagger.login.LoginActivity;
import com.charlye934.loginddagger.login.LoginModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {ApplicationModule.class, LoginModule.class, TwitchModule.class})
public interface ApplicationComponent {
    void inject(LoginActivity target);
}

File: LoginDagger/app/src/main/java/com/charlye934/loginddagger/root/ApplicationComponent.kt
package com.charlye934.loginddagger.root

import com.charlye934.loginddagger.http.TwitchModule
import com.charlye934.loginddagger.login.LoginModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [ApplicationModule::class, LoginModule::class, TwitchModule::class])
interface ApplicationComponent {
    fun inject(target: com.charlye934.loginddagger.login.LoginActivity)
}